[NEXT] Cannot find module vue-i18n.mjs

### Version
@nuxtjs/i18n: 8.0.0-beta.3
nuxt: 3.0.0-rc.13
vue-i18n: "^9.2.2"

### Steps to reproduce
Start nuxi dev

### What is Expected?
Starting dev

### What is actually happening?
Cannot start nuxt:  Cannot find module 'nuxt\node_modules\vue-i18n\dist\vue-i18n.mjs'

--- a/package.json
+++ b/package.json
@@ -79,6 +79,7 @@
     "magic-string": "^0.26.7",
     "mlly": "^1.0.0",
     "pathe": "^1.0.0",
+    "pkg-types": "^1.0.1",
     "ufo": "^1.0.0",
     "unplugin": "^1.0.0",
     "vue-i18n": "^9.3.0-beta.7",

--- a/src/alias.ts
+++ b/src/alias.ts
@@ -1,37 +1,100 @@
-import { resolveModule } from '@nuxt/kit'
-import { pkgModulesDir } from './dirs'
+import createDebug from 'debug'
+import { resolvePath } from '@nuxt/kit'
+import { resolveVueI18nPkgPath, pkgModulesDir } from './dirs'
+import { resolve, parse as parsePath } from 'pathe'
+import { resolveLockfile } from 'pkg-types'
+import { VUE_I18N_PKG, VUE_I18N_BRIDGE_PKG, VUE_ROUTER_BRIDGE_PKG, VUE_I18N_ROUTING_PKG } from './constants'
 
 import type { Nuxt } from '@nuxt/schema'
 
+const debug = createDebug('@nuxtjs/i18n:alias')
+
 export async function setupAlias(nuxt: Nuxt) {
+  const pkgMgr = await getPackageManagerType()
+  debug('setupAlias: pkgMgr', pkgMgr)
+
   // resolve vue-i18@v9
-  const vueI18nPath = nuxt.options.dev ? 'vue-i18n/dist/vue-i18n.mjs' : 'vue-i18n/dist/vue-i18n.runtime.mjs'
-  nuxt.options.alias['vue-i18n'] = resolveModule(vueI18nPath, {
-    paths: nuxt.options.modulesDir
-  })
-  nuxt.options.build.transpile.push('vue-i18n')
+  nuxt.options.alias[VUE_I18N_PKG] = await resolveVueI18nAlias(nuxt)
+  nuxt.options.build.transpile.push(VUE_I18N_PKG)
+  debug('vue-i18n alias', nuxt.options.alias[VUE_I18N_PKG])
 
   // resolve @intlify/shared
-  nuxt.options.alias['@intlify/shared'] = resolveModule('@intlify/shared/dist/shared.mjs', {
-    paths: nuxt.options.modulesDir
-  })
+  nuxt.options.alias['@intlify/shared'] = await resolvePath('@intlify/shared')
   nuxt.options.build.transpile.push('@intlify/shared')
+  debug('@intlify/shared alias', nuxt.options.alias['@intlify/shared'])
 
   // resolve @intlify/vue-router-bridge
-  nuxt.options.alias['@intlify/vue-router-bridge'] = resolveModule('@intlify/vue-router-bridge/lib/index.mjs', {
-    paths: [pkgModulesDir, ...nuxt.options.modulesDir]
-  })
-  nuxt.options.build.transpile.push('@intlify/vue-router-bridge')
+  nuxt.options.alias[VUE_ROUTER_BRIDGE_PKG] = await resolveVueRouterBridgeAlias(pkgModulesDir, pkgMgr)
+  nuxt.options.build.transpile.push(VUE_ROUTER_BRIDGE_PKG)
+  debug('@intlify/vue-router-bridge alias', nuxt.options.alias[VUE_ROUTER_BRIDGE_PKG])
 
   // resolve @intlify/vue-i18n-bridge
-  nuxt.options.alias['@intlify/vue-i18n-bridge'] = resolveModule('@intlify/vue-i18n-bridge/lib/index.mjs', {
-    paths: [pkgModulesDir, ...nuxt.options.modulesDir]
-  })
-  nuxt.options.build.transpile.push('@intlify/vue-i18n-bridge')
+  nuxt.options.alias[VUE_I18N_BRIDGE_PKG] = await resolveVueI18nBridgeAlias(pkgModulesDir, pkgMgr)
+  nuxt.options.build.transpile.push(VUE_I18N_BRIDGE_PKG)
+  debug('@intlify/vue-i18n-bridge alias', nuxt.options.alias[VUE_I18N_BRIDGE_PKG])
 
   // resolve vue-i18n-routing
-  nuxt.options.alias['vue-i18n-routing'] = resolveModule('vue-i18n-routing/dist/vue-i18n-routing.mjs', {
-    paths: [pkgModulesDir, ...nuxt.options.modulesDir]
-  })
-  nuxt.options.build.transpile.push('vue-i18n-routing')
+  nuxt.options.alias[VUE_I18N_ROUTING_PKG] = await resolveVueI18nRoutingAlias(pkgModulesDir, pkgMgr)
+  nuxt.options.build.transpile.push(VUE_I18N_ROUTING_PKG)
+  debug('vue-i18n-routing alias', nuxt.options.alias[VUE_I18N_ROUTING_PKG])
+}
+
+const PackageManagerLockFiles = {
+  'npm-shrinkwrap.json': 'npm-legacy',
+  'package-lock.json': 'npm',
+  'yarn.lock': 'yarn',
+  'pnpm-lock.yaml': 'pnpm'
+} as const
+
+type LockFile = keyof typeof PackageManagerLockFiles
+type _PackageManager = typeof PackageManagerLockFiles[LockFile]
+type PackageManager = _PackageManager | 'unknown'
+
+async function getPackageManagerType(): Promise<PackageManager> {
+  try {
+    const parsed = parsePath(await resolveLockfile())
+    const lockfile = `${parsed.name}${parsed.ext}` as LockFile
+    debug('getPackageManagerType: lockfile', lockfile)
+    if (lockfile == null) {
+      return 'unknown'
+    }
+    const type = PackageManagerLockFiles[lockfile]
+    return type == null ? 'unknown' : type
+  } catch (e) {
+    debug('getPackageManagerType: resolveLockfile error', e)
+    throw e
+  }
+}
+
+async function resolveVueI18nAlias(nuxt: Nuxt) {
+  return resolve(await resolveVueI18nPkgPath(), nuxt.options.dev ? 'dist/vue-i18n.mjs' : 'dist/vue-i18n.runtime.mjs')
+}
+
+async function resolveVueI18nBridgeAlias(pkgModuleDir: string, pkgMgr: PackageManager) {
+  if (pkgMgr === 'npm') {
+    return resolve(pkgModuleDir, `${VUE_I18N_ROUTING_PKG}/node_modules/${VUE_I18N_BRIDGE_PKG}/lib/index.mjs`)
+  } else {
+    const parsed = parsePath(await resolvePath(VUE_I18N_BRIDGE_PKG))
+    return `${parsed.dir}/${parsed.name}.mjs`
+  }
+}
+
+async function resolveVueRouterBridgeAlias(pkgModuleDir: string, pkgMgr: PackageManager) {
+  if (pkgMgr === 'npm') {
+    return resolve(pkgModuleDir, `${VUE_ROUTER_BRIDGE_PKG}/lib/index.mjs`)
+  } else {
+    const parsed = parsePath(await resolvePath(VUE_ROUTER_BRIDGE_PKG))
+    return `${parsed.dir}/${parsed.name}.mjs`
+  }
+}
+
+async function resolveVueI18nRoutingAlias(pkgModuleDir: string, pkgMgr: PackageManager) {
+  if (pkgMgr === 'npm') {
+    return resolve(pkgModuleDir, `${VUE_I18N_ROUTING_PKG}/dist/vue-i18n-routing.mjs`)
+  } else if (pkgMgr === 'pnpm' || pkgMgr === 'yarn') {
+    const parsed = parsePath(await resolvePath(VUE_I18N_ROUTING_PKG))
+    return `${parsed.dir}/dist/vue-i18n-routing.mjs`
+  } else {
+    return await resolvePath(VUE_I18N_ROUTING_PKG)
+  }
 }

--- a/src/constants.ts
+++ b/src/constants.ts
@@ -1,6 +1,10 @@
 import type { NuxtI18nOptionsDefault } from './options'
 
 export const NUXT_I18N_MODULE_ID = '@nuxtjs/i18n' as const
+export const VUE_I18N_PKG = 'vue-i18n' as const
+export const VUE_I18N_BRIDGE_PKG = '@intlify/vue-i18n-bridge' as const
+export const VUE_ROUTER_BRIDGE_PKG = '@intlify/vue-router-bridge' as const
+export const VUE_I18N_ROUTING_PKG = 'vue-i18n-routing' as const
 
 // Options
 const STRATEGY_PREFIX = 'prefix'

--- a/src/dirs.ts
+++ b/src/dirs.ts
@@ -10,6 +10,11 @@ export const runtimeDir = fileURLToPath(new URL('./runtime', import.meta.url))
 const pkgDir = resolve(distDir, '..')
 export const pkgModulesDir = resolve(pkgDir, './node_modules')
 
+debug('distDir', distDir)
+debug('runtimeDir', runtimeDir)
+debug('pkgDir', pkgDir)
+debug('pkgModulesDir', pkgModulesDir)
+
 export async function resolveVueI18nPkgPath() {
   const p = await resolvePath('vue-i18n')
   debug('vue-i18n resolved path', p)

